services/core-api/app/api/document_generation/resources/now_document.py

```diff
@@ -0,0 +1,79 @@
+import os
+import uuid
+from flask import current_app, send_file, request
+from flask_restplus import Resource, fields
+from werkzeug.exceptions import NotFound, BadRequest, NotImplemented
+from app.extensions import api, cache
+from app.api.utils.resources_mixins import UserMixin
+from app.api.utils.access_decorators import requires_role_view_all, requires_role_edit_permit
+from app.api.utils.custom_reqparser import CustomReqparser
+from app.api.constants import TIMEOUT_5_MINUTES, NOW_DOCUMENT_TEMPLATE_TYPES
+
+NOW_DOCUMENT_DOWNLOAD_TOKEN_MODEL = api.model('NoticeOfWorkDocumentDownloadToken',
+                                              {'token_guid': fields.String})
+
+
+def NOW_DOCUMENT_DOWNLOAD_TOKEN(token_guid):
+    return f'now-document-types:download-token:{token_guid}'
+
+
+class NoticeOfWorkDocumentGenerationResource(Resource, UserMixin):
+    parser = CustomReqparser()
+    parser.add_argument('now_application_guid', type=str, location='json', required=True)
+    parser.add_argument('template_data', type=str, location='json', required=True)
+
+    @api.doc(
+        description='Generates the specified document for the NoW using the provided template data.',
+        params={'document_type_code': 'The code indicating the type of document to generate.'})
+    @requires_role_edit_permit
+    def post(self, document_type_code):
+        data = self.parser.parse_args()
+
+        if document_type_code not in NOW_DOCUMENT_TEMPLATE_TYPES:
+            raise NotFound('Document type code not found')
+
+        raise NotImplemented('This endpoint is under construction')
```

Comment:
This is the endpoint that should return a token to the file.

Reply:
I don't think we should modify this POST endpoint's functionality to behave in a way that will be entirely different from how it will behave in the future. I think for the temporary functionality we have now of downloading a docx we need a temporary endpoint, which is what I've created as the GET endpoint.

Reply:
I think that is the POST's behaviour, 
REQUEST: Create me a document 
RESPONSE: Here is a token to get the document you asked for.

Reply:
I see what you mean now. I guess I could pass over the NoW application GUID and an empty template_data object even though it's not being used at the moment.